### PR TITLE
enhancement(reduce): Flushes all reduce state based on condition

### PR DESCRIPTION
**Pull request by dbcfd**
Adds the ability to flush all reduce state when a condition is hit, such as a process ending.

### CODE REVIEW

File: src/transforms/reduce/mod.rs [hunk 1]
```diff
@@ -40,6 +40,7 @@ pub struct ReduceConfig {
     /// reduce.
     pub ends_when: Option<AnyCondition>,
     pub starts_when: Option<AnyCondition>,
+    pub flushes_when: Option<AnyCondition>,
```
Comment (dbcfd) [inline src/transforms/reduce/mod.rs:43]: I feel like this name conflicts with the `flushes_after` parameter, but not sure of what a better name would be.

**Comment by jszwedko** [comment]
@lukesteensen Bruce mentioned you'd probably be a better reviewer here.

**Comment by neuronull** [comment]
Hi @dbcfd !
Apologies for the delay in getting this PR reviewed.
Are you still interested in making the contribution?
If so we can provide aid to progress it's inclusion.

**Comment by tobz** [comment]
Given the overall lack of activity, we're going to close this as stale.  Feel free to pick it up back if it becomes relevant again for your in the future.

In general, though, we'd likely want to see a little more separation between the flushing behavior and whatever the intent was with the batching configuration. Given that the reduce transform is inherently batching events together, it's not clear what else you would be batching on or for. The flushing behavior, though -- flushing _everything_ instead of just a single merge state -- is straightforward, though.

**Pull request closed without merging**